model/modelpb/context.go

```diff
@@ -17,13 +17,12 @@
 
 package modelpb
 
+import "github.com/elastic/apm-data/model/common"
+
 // updateFields transforms in, returning a copy with sanitized keys,
 // suitable for storing as "custom" in transaction and error documents.
-func updateFields(in map[string]any) {
-	for k, v := range in {
-		if lk := sanitizeLabelKey(k); lk != k {
-			delete(in, k)
-			in[lk] = v
-		}
+func updateFields(in []*common.KeyValue) {
```

Comment:
Could we do this sanitisation inline in the MarshalFastJSON methods in modeljson?

Reply:
Not all of the key values seem to be sanitized so that would change behaviour

Reply:
Not all `[]*modelpb.KeyValue`s are sanitized. If we apply the sanitisation inside the marshaling method it would apply to every field.

Reply:
I see, thanks for explaining. We could address that with a named type, like `type SanitizedKeyValues []*KeyValue`, with a custom `MarshalFastJSON` method. What you have implemented is fine for now though.